src/datamodel/validator/Uuid.cpp

```diff
@@ -10,3 +10,134 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/validator/Uuid.h"
+
+using namespace std::chrono_literals;
+
+namespace {
+std::chrono::system_clock::time_point getUuidTimestamp(
+    const uprotocol::v1::UUID& uuid) {
+	uint64_t msb = uuid.msb();
+	uint64_t timestamp =
+	    (msb >> 16) & 0xFFFFFFFFFFFF;  // Extract 48-bit timestamp
+	return std::chrono::system_clock::time_point(
+	    std::chrono::milliseconds(timestamp));
+}
+}  // namespace
+
+namespace uprotocol::datamodel::validator::uuid {
+
+std::string_view message(Reason reason) {
+	switch (reason) {
+		case Reason::WRONG_VERSION:
+			return "Invalid UUID version";
+		case Reason::UNSUPPORTED_VARIANT:
+			return "Unsupported UUID variant";
+		case Reason::FROM_THE_FUTURE:
+			return "UUID timestamp is from the future";
+		case Reason::EXPIRED:
+			return "UUID has expired";
+		default:
+			return "Unknown reason";
+	}
+}
+
+ValidationResult isUuid(const uprotocol::v1::UUID uuid) {
+	uint64_t msb = uuid.msb();
+	uint64_t lsb = uuid.lsb();
+
+	uint8_t version = (msb >> 12) & 0xF;
```

Comment:
You've got this same code several places.  It might be easier to read if you created an _internal_getVersion that only does the shifting/bitmasking, but no additional checks, then both isUuid and getVersion could both call the new function.